### PR TITLE
added support for future date, changed the action to run on a schedul…

### PR DESCRIPTION
**Pull request by dpmcnevin**
…e, and added new episode

### CODE REVIEW

File: eleventy-config.js [hunk 1]
```diff
@@ -34,13 +35,26 @@ function notIndex(thing) {
   return !thing.inputPath.includes('index.njk');
 }
 
+function currentEpisodes(episode) {
+  let episode_date = episode && episode.data && episode.data.stuff && episode.data.stuff.date
```
Comment (jimkang) [inline eleventy-config.js]: https://www.npmjs.com/package/get-at-path

File: reading-list.njk [hunk 2]
```diff
@@ -8,11 +8,13 @@ pageHeading: Reading List
   <h3>Season {{ season.number }} - <u>{{ season.title }}</u> by {{ season.author }}</h3>
   <ul>
   {%- for episode in season.episodes -%}
+    {% set ep = episode | getEpisodeEntry(season, collections.episodes) %}
```
Comment (jimkang) [inline reading-list.njk]: Can you name `ep` to `releasedEpisode` or something like that?

File: eleventy-config.js [hunk 3]
```diff
@@ -67,4 +81,12 @@ function getReadingListEntry(ep, readingList) {
   return { season, episode };
 }
 
+function getEpisodeEntry(rl, rlSeason, episodeList) {
```
Comment (jimkang) [inline eleventy-config.js]: What does `rl` mean here? It seems like the episode you want. Naming the function `getEpisodeEntryFromList` might make things clearer, too.

File: eleventy-config.js [hunk 4]
```diff
@@ -34,13 +35,26 @@ function notIndex(thing) {
...
 
+function currentEpisodes(episode) {
+  let episode_date = episode && episode.data && episode.data.stuff && episode.data.stuff.date
+
+  // If the episode defines a date and it's after current hide it
+  if (episode_date && episode_date > new Date()) {
+    return false;
+  }
+
+  return true;
```
Comment (jimkang) [inline eleventy-config.js:47]: Should probably hide it if it's missing a date on anything else, too.

**Review by jimkang — CHANGES_REQUESTED** [review]
The approach makes sense. I guess you'd test this after merging?

File: eleventy-config.js (hunk 4 above)
Comment (dpmcnevin) [reply eleventy-config.js:47]: I actually wanted to "fail open" in that case.. only if you specify a date does it check to see if that date is after now

File: eleventy-config.js (hunk 3 above)
Comment (dpmcnevin) [reply eleventy-config.js]: readling list.. the first parameter to the filter function what's piped into it, so I guess it'd actually be like readingListEntry

File: eleventy-config.js (hunk 1 above)
Comment (dpmcnevin) [reply eleventy-config.js]: I was going to say that we shouldn't pull in external dependencies if we don't have to, but then i noticed it was from you, so i guess you can mine bitcoin in either repo if you wanted to

File: reading-list.njk (hunk 2 above)
Comment (dpmcnevin) [reply reading-list.njk]: yep

**Comment by dpmcnevin** [comment]
> The approach makes sense. I guess you'd test this after merging?

Yeah, i guess.. I checked it locally by messing with the date on the new episode and it didn't show up when i didn't want it to. I am more unsure about the scheduled action, but we'll just have to see if it works

File: eleventy-config.js (hunk 1 above)
Comment (jimkang) [reply eleventy-config.js]: Heh, this whole thing was a scheme to get bitcoin mined on your computer!

You can also just copy in the source. I think it's ten lines.

File: eleventy-config.js (hunk 3 above)
Comment (jimkang) [reply eleventy-config.js]: Ah, maybe this function should be `getEpisodeForReadingListEntry`, then?

File: eleventy-config.js (hunk 4 above)
Comment (jimkang) [reply eleventy-config.js:47]: I guess that could work, though I'm not sure what a podcast client would do with that.

**Review by jimkang — APPROVED** [review]

**Pull request merged**